providers/base/bin/network.py

```diff
@@ -166,27 +167,71 @@ def summarize_cpu(self):
                 avg_cpu = sum_cpu / n
         return avg_cpu
 
+    def find_numa(self, device):
+        """Return the NUMA node of the specified network device."""
+        filename = "/sys/class/net/" + device + "/device/numa_node"
+        try:
+            file = open(filename, "r")
+            node_num = int(file.read())
```

Comment:
This would be a bit cleaner using with:
try:
  with open(filename, "r") as file:
    node_num = int(file.read())
except FileNotFoundError